docs/syntax.md

```diff
@@ -0,0 +1,340 @@
+# WireViz Syntax
+
+## Main sections
+
+```yaml
+connectors:  # dictionary of all used connectors
+  <str>   :    # unique connector designator/name
+    ...          # connector attributes (see below)
+  <str>   :
+    ...
+  ...
+
+cables:      # dictionary of all used cables and wires
+  <str>   :    # unique cable designator/name
+    ...          # cable attributes (see below)
+  <str>   :
+    ...
+  ...
+
+connections:  # list of all connections to be made
+              # between cables and connectors
+  -
+    ...         # connection set (see below)
+  -
+    ...
+  ...
+
+additional_bom_items:  # custom items to add to BOM
+  - <bom-item>           # BOM item (see below)
+  ...
+
+```
+
+## Connector attributes
+
+```yaml
+<str>   :  # unique connector designator/name
```

Comment:
This is probably a very deep rabbit hole, with all kinds of exceptions:
- numbers are accepted, but need to go in quotes so they are parsed as strings
- some special YAML characters (`-`, `:` and others) can be used by putting the string in quotes
- some characters will be accepted, but
  - might not show up in the graphical output
  - might mess with the BOM export
- some obscure cases might throw  errors

Honestly for now I would leave the explanation as-is, and let the user figure it out.
I do agree that it merits a more detailed description, but I would leave that for a later date.

Reply:
As this connector designator (and similar for cables) currently is used directly to name the Graphviz node, it must be case insensitive unique within the union of connectors and cables in the harness to avoid reference mix-ups similar to those discovered in #160.

Reply:
I would like to leave it as-is now, hopefully fix #160 soon and then add "case-sensitive" to the syntax description